File: dietideals-service/src/main/java/it/uninastudents/dietidealsservice/model/entity/Utente.java
package it.uninastudents.dietidealsservice.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import it.uninastudents.dietidealsservice.model.base.BaseEntity;
import it.uninastudents.dietidealsservice.model.entity.enums.RuoloUtente;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
@Entity
@Table(name = "utente", uniqueConstraints = {
        @UniqueConstraint(columnNames = "email"),
        @UniqueConstraint(columnNames = "username"),
        @UniqueConstraint(columnNames = "idAuth")
})
public class Utente extends BaseEntity {

    private String idAuth;

    @Column(nullable = false)
    private String username;

    @Column(nullable = true)
    private RuoloUtente ruolo;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false)
    private String cognome;

    @Column(nullable = false)
    private LocalDate dataNascita;

    @Column(nullable = false)
    private String email;

    @Column(length = 300)
    private String descrizione;

    private String facebook;

    private String instagram;

    private String twitter;

    private String sitoWeb;

    private String indirizzo;

    private String partitaIva;

    private String urlDocumentoIdentita;

    private String urlFotoProfilo;

    @EqualsAndHashCode.Exclude
    @JsonIgnore
    @OneToOne(cascade = CascadeType.ALL)
    private ContoCorrente contoCorrente;

    @EqualsAndHashCode.Exclude
    @JsonIgnore
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Offerta> offerte;

    @EqualsAndHashCode.Exclude
    @JsonIgnore
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Asta> aste;

    @EqualsAndHashCode.Exclude
    @JsonIgnore
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Notifica> notifiche;

    @EqualsAndHashCode.Exclude
    @JsonIgnore
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Carta> carte;

}

File: dietidealsapp/app/src/main/java/com/example/dietideals_app/model/Utente.kt
package com.example.dietideals_app.model

import com.example.dietideals_app.model.enum.RuoloUtente
import java.io.File
import java.time.LocalDate
import java.util.UUID

class Utente(
    val id: UUID,
    val username: String,
    val nome: String,
    val cognome: String,
    val dataDiNascita: LocalDate,
    var partitaIva: String,
    var documentoRiconoscimento: File,
    var immagineProfilo: File,
    var contoCorrente: ContoCorrente?,
    var carteAssociate: MutableSet<Carta> = mutableSetOf(),
    var aste: MutableSet<Asta> = mutableSetOf(),
    var offerte: MutableSet<Offerta> = mutableSetOf(),
    var notifiche: MutableSet<Notifica> = mutableSetOf(),
    var ruolo: RuoloUtente,
    var descrizione: String?,
    var sitoWeb: String?,
    var indirizzo: String?,
    var instagram: String?,
    var facebook: String?,
    var twitter: String?
)